packages/frontend/app/routes/peers.$peerId.tsx

```diff
@@ -217,20 +238,22 @@ export default function ViewPeerPage() {
         {/* Peer Asset Info - END */}
         {/* DELETE PEER - Danger zone */}
         <DangerZone title='Delete Peer'>
-          <Form method='post'>
+          <Form method='post' onSubmit={submitHandler}>
             <Input type='hidden' name='id' value={peer.id} />
-            <Button
-              type='submit'
-              intent='danger'
-              name='intent'
-              value='delete'
-              aria-label='delete peer'
-            >
+            <Input type='hidden' name='intent' value='delete' />
+            <Button type='submit' intent='danger' aria-label='delete peer'>
               Delete peer
             </Button>
           </Form>
         </DangerZone>
       </div>
+      <ConfirmationDialog
+        ref={dialogRef}
+        onConfirm={onConfirm}
+        title='Delete Peer'
+        keyword='delete peer'
```

Comment:
```suggestion
        keyword={peer.name || 'delete peer'}
```
I think that kind of follows how archiving & deleting repos in Github (as an example).
Too bad `peer.name` is optional...

Reply:
Yes, that was the pattern I was going for in the first place, but like you mentioned I noticed that `peer.name` is optional.